### ISSUE TITLE
Installation issue: py-kiwisolver (macOS + gcc)

### ISSUE THREAD

**Issue by nilsvu** [LLNL/spack#28522]
### Steps to reproduce the issue

Installation fails on macOS with GCC:

```spack install py-kiwisolver%gcc platform=darwin```

Error: GCC doesn't have a `-stdlib` option. See upstream issue: https://github.com/nucleic/kiwi/issues/126

### Information on your system

* **Spack:** 0.17.1-917-1a5add3021
* **Python:** 3.9.9
* **Platform:** darwin-monterey-skylake
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by adamjstewart** [comment]
Out of curiosity, how did you install Python? Even the Python package doesn't build on macOS with GCC. I would highly recommend avoiding GCC on macOS if possible, many packages will only build with Apple Clang.

**Comment by nilsvu** [comment]
I installed Python with Homebrew :) also opened the related https://github.com/spack/spack/issues/28528

**Comment by adamjstewart** [comment]
As a fellow macOS user, I would love to see better macOS testing and support. Personally I wouldn't recommend using anything other than Apple Clang on macOS, but if you convince any developers to fix their software with macOS + GCC, we can add as many patches as you can find.

**Comment by sethrj** [comment]
Yeah, installing with `%gcc` on mac has become increasingly difficult over the years, in part due to the decoupling of the macOS system version and the SDK toolchains installed by XCode (and by CommandLineTools).
